### PR TITLE
don't send non-zero action values when spike did not proc

### LINKED ISSUE #4937: 🐛 item spikes ignoring proc rate
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)
n/a

## Branch affected by issue 
`base`

## Steps to reproduce
Items that give spike effects are procing regardless of the die roll for their chance. The logic allows it to get past some checks and then assumes a non proc wouldn't have made it into said die roll, so it winds up going past with the default values instead of sending back the 3 zeroes that are expected. Downstream people are making their own fixes for this **without reporting it.**

## Expected behavior
Proc rates obeyed.

### PR DESCRIPTION
**Pull request by TeoTwawki**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
The entire program flow for handling spikes in general (not just equipment with spikes on them) is flat out wrong. It assumes we will be doing damage and an animation of some kind, before we even know if we've got an effect or gear that succeeded a "roll" then overwrites them with whatever values each conditional branch actually needed when its wrong, and at the end of none were entered finally overwrites with zero. this resulted in an oversight in spikes gear where a condition is always met so it never reaches those zero values at the end. The quick band-aid is to add another set of overwriting with zeroes there.

I first looked at correcting teh program flow, but its a lot of refactoring for something that is getting tirn down: we're moving how we handle damage over to lua in the future, and spikes were planned to move there even before that, so a full rewrite in cpp doesn't make sense when its going to lua later anyway.

This PR also includes some additional message enum work, and a bunch of format cleanup.

This closes #4937

## Steps to test these changes
Equip an item with spikes, let a bird in sky beat on you for an hr. see that sometimes they proc and sometimes they don't. Change the rate in the db on the item, restart server and retest and see that the rate MATTERS now.

Try genie weskit and see that curse message displays properly (its been missing its enum for years).

### CODE REVIEW

File: scripts/enum/msg.lua [hunk 1]
```diff
@@ -405,3 +406,4 @@ xi.msg.system =
     TRUST_NO_ENMITY              = 300, -- You cannot use Trust magic while having gained enmity.
     TRUST_SOLO_OR_LEADER         = 301, -- You cannot use Trust magic unless you are solo or the party leader.
 }
+return xi.msg
```
Comment (Xaver-DaRed) [inline scripts/enum/msg.lua]: I dont see returns in enum files? Is this return needed?

Comment (TeoTwawki) [reply scripts/enum/msg.lua]: accidental

Comment (TeoTwawki) [reply scripts/enum/msg.lua]: during testing log would whine about the cache, this was just to shut it the F up, and then I forgot it

**Review by Xaver-DaRed — APPROVED** [review]
CI hit you hard, but ignoring the CI style checks and the accidental return, everything looks ok

**Comment by TeoTwawki** [comment]
> CI hit you hard, but ignoring the CI style checks and the accidental return, everything looks ok

misaligned stuff I didn't write in file I touched /sulk

**Comment by TeoTwawki** [comment]
fixed Ci issues and the accidental return

**Pull request merged**